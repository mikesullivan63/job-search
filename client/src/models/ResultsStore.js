import { types, detach, destroy } from "mobx-state-tree";
import Job from "./Job";
import Board from "./Board";
import User from "./User";
import { objectComparator } from "../util/comparator";

const ResultsStore = types
  .model("ResultsStore", {
    user: User,
    loggedIn: types.boolean,
    activeJobs: types.array(Job),
    ignoredJobs: types.array(Job),
    searchResults: types.array(Board)
  })
  .actions(self => ({
    setActiveJobs(jobs) {
      self.activeJobs = jobs;
    },
    addActiveJob(job) {
      self.activeJobs.push(job);
    },
    archiveActiveJob(jobId) {
      let temp = self.activeJobs.find(el => el._id === jobId);
      self.ignoredJobs.push(detach(temp));
    }
  }))
  .actions(self => ({
    setIgnoredJobs(jobs) {
      self.ignoredJobs = jobs;
    },
    addIgnoredJob(job) {
      self.ignoredJobs.push(job);
    }
  }))
  .actions(self => ({
    setSearchResults(boards) {
      self.searchResults.replace(
        boards.slice().sort(objectComparator("company"))
      );
    },
    addSearchResult(board) {
      //Mutable
      let loc = self.searchResults.findIndex(
        el => el.company === board.company
      );
      let temp = self.searchResults[loc];
      self.searchResults[loc] = board;
      destroy(temp);
    }
  }))
  .actions(self => ({
    setUser(user, token) {
      user.token = token.token;
      console.log("Saving user: " + JSON.stringify(user, null, 2));
      self.user = user;
      self.loggedIn = true;
    },
    logout() {
      self.loggedIn = false;
      destroy(self.user);
    }
  }))
  .views(self => ({
    isIgnored(job) {
      return self.ignoredJobs.some(el => el.url === job.url);
    },
    isActive(job) {
      return self.activeJobs.some(el => el.url === job.url);
    },
    getActiveJob(job) {
      return self.activeJobs.find(el => el.url === job.url);
    },
    getIgnoredJob(job) {
      return self.ignoredJobs.find(el => el.url === job.url);
    },
    getSearchResults() {
      return self.searchResults.slice();
    },
    getUser() {
      return self.user;
    },
    isLoggedIn() {
      return self.loggedIn;
    }
  }));

export default ResultsStore;
