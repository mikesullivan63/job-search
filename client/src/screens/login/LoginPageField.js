import React from "react";
import { Form, Input } from "semantic-ui-react";

class LoginPageField extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Form.Field
        fluid
        control={Input}
        iconPosition="left"
        inline="true"
        id={"login-form-input-control" + this.props.name}
        name={this.props.name}
        type={this.props.type}
        icon={this.props.icon}
        label={this.props.label}
        placeholder={this.props.placeholder}
        error={
          this.props.errors && this.props.errors.length > 0
            ? this.props.errors
            : null
        }
        onChange={this.props.handleChange}
      />
    );
  }
}
export default LoginPageField;
