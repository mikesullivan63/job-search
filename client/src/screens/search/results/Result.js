import React from "react";
import PendingResult from "./PendingResult";

const Result = props => {
  //  if (props.company.status === "PENDING") {
  return <PendingResult key={props.company.company} company={props.company} />;
  //}
};

export default Result;
