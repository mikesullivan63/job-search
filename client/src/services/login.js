//Pass in reference to MobX-State-Tree store to allow for simpler manipultion
function setStore(localStore) {
  this.store = localStore;
}

//Methods to handle storing users while logged in
function storeUser(user, token) {
  this.store.setUser(user, token.token);
  localStorage.setItem("currentUser", JSON.stringify(token));
}

function updateProfile(user) {
  this.store.setUser(user, this.store.getUser().token);
}

function logout() {
  // remove user from local storage to log user out
  localStorage.removeItem("currentUser");
  this.store.logout();
}

function getUser() {
  return this.store.getUser();
}

export const loginService = {
  setStore,
  logout,
  storeUser,
  updateProfile,
  getUser
};
