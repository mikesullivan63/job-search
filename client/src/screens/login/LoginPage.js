import React from "react";
import { Redirect } from "react-router-dom";
import { observer } from "mobx-react";
import ProfileForm from "../common/ProfileForm";
import { authenticationService } from "../../services/authentication";
import { Grid, Message } from "semantic-ui-react";

class LoginPage extends ProfileForm {
  constructor(props) {
    super(props);

    this.state.fields = [
      {
        name: "email",
        label: "E-mail address",
        icon: "user"
      },
      {
        name: "password",
        label: "Password",
        icon: "lock",
        type: "password"
      }
    ];
  }

  submitForm = () => {
    return authenticationService.login(
      this.getValueForField("email"),
      this.getValueForField("password")
    );
  };

  handleSuccess = result => {
    console.log("Logged user in");
  };

  render() {
    if (this.props.store.isLoggedIn()) {
      return <Redirect to="/" />;
    }

    return (
      <Grid
        textAlign="center"
        style={{ height: "100vh" }}
        verticalAlign="middle"
      >
        <Grid.Column style={{ maxWidth: 450 }}>
          {this.renderHeaderAndForm(
            "login",
            "Log-in to your account",
            "Login",
            "Logged in successfully",
            "Error attempting to log in"
          )}
          <Message>
            New to us? <a href="/register">Sign Up</a>
          </Message>
        </Grid.Column>
      </Grid>
    );
  }
}

export default observer(LoginPage);
