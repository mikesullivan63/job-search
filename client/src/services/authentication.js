import { loginService } from "./login";
import { handleResponse } from "./handleResponse";

function login(data) {
  return new Promise((resolve, reject) => {
    fetch("/user/login", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(data)
    })
      .then(response => handleResponse.handlePrivateResponse(response))
      .then(token => {
        fetch("/user/profile", {
          headers: {
            "Content-Type": "application/json",
            Authorization: `Bearer ${token.token}`
          }
        })
          .then(response => handleResponse.handlePrivateResponse(response))
          .then(profile => {
            loginService.storeUser(profile, token);
            resolve(token);
          })
          .catch(error => {
            console.log("Error loading user profile", error);
            loginService.logout();
            reject(error);
          });
      })
      .catch(error => {
        console.log("Error logging user in", error);
        loginService.logout();
        reject(error);
      });
  });
}

function remember() {
  return new Promise((resolve, reject) => {
    const persistentToken = localStorage.getItem("currentUser");
    if (persistentToken && JSON.parse(persistentToken).token) {
      fetch("/user/profile", {
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${JSON.parse(persistentToken).token}`
        }
      })
        .then(response => handleResponse.handlePrivateResponse(response))
        .then(profile => {
          loginService.storeUser(profile, JSON.parse(persistentToken));
          return resolve(profile);
        })
        .catch(error => {
          console.log("Error loading user profile", error);
          loginService.logout();
          return reject(error);
        });
    } else {
      loginService.logout();
      return resolve(null);
    }
  });
}

function authHeader(otherHeaders) {
  // return authorization header with jwt token
  const currentUser = loginService.getUser();
  if (currentUser && currentUser.token) {
    return {
      ...otherHeaders,
      ...{ Authorization: `Bearer ${currentUser.token}` }
    };
  } else {
    return otherHeaders;
  }
}

export const authenticationService = {
  login,
  remember,
  authHeader
};
