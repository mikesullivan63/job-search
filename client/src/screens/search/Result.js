import React from "react";
import { observer } from "mobx-react";
import styled from "styled-components";
import { Grid, Cell } from "react-foundation";
import { authenticationService } from "../../services/authentication";

const StyledResultsCompanyName = styled.h4`
  margin: 0.1em;
`;

const StyledResultsCompanyJobTitle = styled.span`
  vertical-align: top;
`;
const StyledResultsCompanyJobLocation = styled.span`
  vertical-align: top;
`;

/*
    Object coming in should look like this:
    {
        company: Company Name
        jobs: [{
            title: Job Title,
            location: Job Location,
            url: URL to posting
        }]
    }
*/

const addJobToList = function(event, data, url, updateCallback) {
  event.preventDefault();

  fetch(url, {
    method: "POST",
    headers: {
      ...authenticationService.authHeader(),
      ...{ "Content-Type": "application/json" }
    },
    body: JSON.stringify(data)
  })
    .then(res => res.json())
    .then(res => updateCallback(res))
    .catch(reason => console.log("Error: " + reason));
};

function ArchiveLink(props) {
  const archiveLink = (event, jobId) =>
    addJobToList(event, { jobId }, "/job/archive-job", result => {
      this.props.store.setActiveJobs(result.active);
      this.props.store.setIgnoredJobs(result.ignored);
    });

  return (
    <button
      className="jobToggleButton"
      onClick={event => {
        props.archiveLink(event, props.job._id);
      }}
    >
      Archive
    </button>
  );
}

function WatchLink(props) {
  const watchLink = (event, board, url, title, location) =>
    addJobToList(
      event,
      { board, url, title, location },
      "/job/add-job",
      result => this.propp.store.setActiveJobs(result)
    );

  return (
    <button
      className="jobToggleButton"
      onClick={event => {
        props.watchCallback(
          event,
          props.company,
          props.job.url,
          props.job.title,
          props.job.location
        );
      }}
    >
      Watch
    </button>
  );
}

function IgnoreLink(props) {
  const ignoreLink = (event, board, url, title, location) =>
    addJobToList(
      event,
      { board, url, title, location },
      "/job/ignore-job",
      result => this.propp.store.setIgnoredJobs(result)
    );

  return (
    <button
      className="jobToggleButton"
      onClick={event => {
        props.ignoreCallback(
          event,
          props.company,
          props.job.url,
          props.job.title,
          props.job.location
        );
      }}
    >
      Ignore
    </button>
  );
}

function ResultsCompanyJob(props) {
  return (
    <Grid className="display">
      <Cell small={4} medium={3} large={4}>
        <StyledResultsCompanyJobTitle>
          <a href={props.job.url} target="_blank" rel="noopener noreferrer">
            {props.job.title}
          </a>
        </StyledResultsCompanyJobTitle>
      </Cell>
      <Cell small={4} medium={3} large={4}>
        <StyledResultsCompanyJobLocation>
          {props.job.location}
        </StyledResultsCompanyJobLocation>
      </Cell>
      <Cell small={2} medium={1} large={1}>
        {props.job.isActive && (
          <ArchiveLink store={props.store} job={props.job} />
        )}
        {!props.job.isActive && (
          <WatchLink
            store={props.store}
            job={props.job}
            company={props.company.company}
          />
        )}
      </Cell>
      <Cell small={2} medium={1} large={1}>
        {!props.job.isActive && (
          <IgnoreLink
            store={props.store}
            job={props.job}
            company={props.company.company}
          />
        )}
      </Cell>
    </Grid>
  );
}

function Result(props) {
  let displayStyle = "callout";
  let displayMessage = "";

  if (
    props.company.state === "PENDING" ||
    props.company.state === "" ||
    !props.company.state
  ) {
    displayStyle = "callout";
    displayMessage = "Pending Lookup";
  } else if (props.company.state === "ERROR") {
    displayStyle = "callout alert";
    displayMessage = "Error: " + props.company.error.substring(0, 100);
  } else if (props.company.state === "COMPLETED") {
    displayStyle =
      "callout " +
      (!props.company.jobs || props.company.jobs.length === 0
        ? "secondary"
        : "primary");
    displayMessage =
      !props.company.jobs || props.company.jobs.length === 0
        ? "No Matching Jobs"
        : "";
  }

  console.log("Displaying list");

  let jobsToDisplay = props.company.jobs
    .filter(el => !this.props.store.isIgnored(el))
    .map(el => {
      el.isActive = this.props.store.isActive(el);
      return el;
    })
    .sort();

  return (
    <div className={displayStyle}>
      <Grid className="display">
        <Cell small={6} medium={4} large={4}>
          <a href={props.company.url} target="_blank" rel="noopener noreferrer">
            <StyledResultsCompanyName>
              {props.company.company}
            </StyledResultsCompanyName>
          </a>
        </Cell>
        {displayMessage !== "" && (
          <Cell small={6} medium={6} large={4} className="result-message">
            {displayMessage}
          </Cell>
        )}
      </Grid>
      {jobsToDisplay.map(job => {
        return (
          <ResultsCompanyJob
            store={props.store}
            job={job}
            company={props.company}
          />
        );
      })}
    </div>
  );
}

export default observer(Result);
