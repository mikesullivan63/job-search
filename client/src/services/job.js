import { authenticationService } from "./authentication";

//Pass in reference to MobX-State-Tree store to allow for simpler manipultion
let store = null;
function setStore(localStore) {
  this.store = localStore;
}
function addJobToList(data, url, updateCallback) {
  fetch(url, {
    method: "POST",
    headers: authenticationService.authHeader({
      "Content-Type": "application/json"
    }),
    body: JSON.stringify(data)
  })
    .then(res => res.json())
    .then(res => updateCallback(res))
    .catch(reason => console.log("Error: " + reason));
}

function ignoreJob(board, url, title, location) {
  addJobToList({ board, url, title, location }, "/job/ignore-job", result =>
    this.store.addIgnoredJob(result.find(el => el.url === url))
  );
}

function watchJob(board, url, title, location) {
  addJobToList({ board, url, title, location }, "/job/add-job", result =>
    this.store.addActiveJob(result.find(el => el.url === url))
  );
}

function archiveJob(jobId) {
  addJobToList({ jobId }, "/job/archive-job", result => {
    this.store.archiveActiveJob(jobId);
  });
}

function watchIgnoredJob(jobId) {
  addJobToList({ jobId }, "/job/watch-ignore-job", result => {
    this.store.addActiveJobFromIgnored(jobId);
  });
}

export const jobService = {
  setStore,
  ignoreJob,
  watchJob,
  archiveJob,
  watchIgnoredJob
};
