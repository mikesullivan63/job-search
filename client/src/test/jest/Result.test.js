import React from "react";
//import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import { mount, renderer } from "enzyme";
import Result from "../../screens/search/results/Result";
import ResultsStore from "../../models/ResultsStore";
import { jobService } from "../../services/job";
import { loginService } from "../../services/login";

//semaphoreci.com/community/tutorials/how-to-test-react-and-mobx-with-jest

describe("Result Display tests", () => {
  Enzyme.configure({ adapter: new Adapter() });

  const buildStore = () => {
    const store = ResultsStore.create({});
    store.setActiveJobs([
      {
        _id: "ABC123",
        title: "Job Title ABC123",
        location: "San Francisco",
        url: "https://www.example.com/job/ABC123"
      },
      {
        _id: "ABC456",
        title: "Job Title ABC456",
        location: "San Francisco",
        url: "https://www.example.org/job/ABC456"
      }
    ]);

    store.setIgnoredJobs([
      {
        _id: "DEF123",
        title: "Job Title DEF123",
        location: "San Francisco",
        url: "https://www.example.com/job/DEF123"
      },
      {
        _id: "DEF456",
        title: "Job Title DEF456",
        location: "San Francisco",
        url: "https://www.example.org/job/DEF456"
      }
    ]);

    store.setSearchResults([
      {
        company: "Example.com",
        url: "https://www.example.com/job/",
        status: "COMPLETED",
        error: "",
        jobs: [
          {
            _id: "ABC123",
            title: "Job Title ABC123",
            location: "San Francisco",
            url: "https://www.example.com/job/ABC123"
          },
          {
            _id: "DEF123",
            title: "Job Title DEF123",
            location: "San Francisco",
            url: "https://www.example.com/job/DEF123"
          },
          {
            _id: "",
            title: "Job Title GHI123",
            location: "San Francisco",
            url: "https://www.example.com/job/GHI123"
          },
          {
            _id: "",
            title: "Job Title JKL123",
            location: "San Francisco",
            url: "https://www.example.com/job/JKL123"
          }
        ],
        otherJobs: [
          {
            _id: "",
            title: "Job Title MNO123",
            location: "San Francisco",
            url: "https://www.example.com/job/MNO123"
          }
        ]
      },
      {
        company: "Example.org",
        url: "https://www.example.org/job/",
        status: "COMPLETED",
        error: "",
        jobs: [
          {
            _id: "ABC456",
            title: "Job Title ABC456",
            location: "San Francisco",
            url: "https://www.example.org/job/ABC456"
          },
          {
            _id: "DEF456",
            title: "Job Title DEF456",
            location: "San Francisco",
            url: "https://www.example.org/job/DEF456"
          }
        ]
      },
      {
        company: "Example.io",
        url: "https://www.example.io/job/",
        status: "PENDING",
        error: ""
      },
      {
        company: "Error.com",
        url: "https://www.error.com/job/",
        status: "ERROR",
        error: "Problem accessing page"
      }
    ]);

    loginService.setStore(store);
    jobService.setStore(store);
    return store;
  };

  test("Render result for pending result", () => {
    const store = buildStore();
    const company = store.searchResults.find(el => el.status === "PENDING");

    const component = mount(
      <Result store={store} key={company.company} company={company} />
    );

    expect(component).toMatchSnapshot();
  });

  test("Render result for error result", () => {
    const store = buildStore();
    const company = store.searchResults.find(el => el.status === "ERROR");

    const component = mount(
      <Result store={store} key={company.company} company={company} />
    );

    expect(component).toMatchSnapshot();
  });

  test("Render result for completed results", done => {
    const store = buildStore();
    const company = store.searchResults.find(el => el.status === "COMPLETED");
    const component = mount(
      <Result store={store} key={company.company} company={company} />
    );
    expect(component.find("div.activeJobs a").length).toBe(3); //3 'Active' jobs
    expect(component.find("div.ignoredJobs a").length).toBe(1); //1 'Ignored' job
    expect(component.find("div.otherJobs a").length).toBe(1); //1 'Other' job

    expect(component.find("button.archiveButton").length).toBe(1);
    expect(component.find("button.watchButton").length).toBe(2);
    expect(component.find("button.ignoreButton").length).toBe(2);
    expect(component.find("button.watchIgnoredButton").length).toBe(1);

    expect(component).toMatchSnapshot();
    done();
  });

  test("Render result for good results, archive one of them", done => {
    jest.restoreAllMocks();
    jest.spyOn(global, "fetch").mockImplementation(() => {
      return new Promise((resolve, reject) => {
        resolve(new Response('"OK"'));
      });
    });

    const store = buildStore();
    const company = store.searchResults.find(el => el.status === "COMPLETED");
    const component = mount(
      <Result store={store} key={company.company} company={company} />
    );
    expect(store.activeJobs.length).toBe(2);
    expect(store.ignoredJobs.length).toBe(2);

    expect(component.find("div.completedResult").length).toBe(1);
    expect(component.find("div.activeJobs a").length).toBe(3); //3 'Active' jobs
    expect(component.find("div.ignoredJobs a").length).toBe(1); //1 'Ignored' job
    expect(component.find("div.otherJobs a").length).toBe(1); //1 'Other' job

    expect(component.find("button.archiveButton").length).toBe(1);
    component.find("button.archiveButton").simulate("click");

    setTimeout(() => {
      component.update();

      expect(store.activeJobs.length).toBe(1);
      expect(store.ignoredJobs.length).toBe(3);
      expect(component.find("button.archiveButton").length).toBe(0);
      expect(component.find("div.activeJobs a").length).toBe(2); //2 'Active' jobs remaining
      expect(component.find("div.ignoredJobs a").length).toBe(2); //2 'Ignored' job, as we just added 0ne
      expect(component.find("div.otherJobs a").length).toBe(1); //1 'Other' job, unchanged

      expect(component).toMatchSnapshot();

      expect(global.fetch).toHaveBeenCalledTimes(1);
      expect(global.fetch).toHaveBeenCalledWith(
        "/job/archive-job",
        expect.anything()
      );
      global.fetch.mockClear();

      done();
    });
  });

  test("Render result for good results, watch one", done => {
    jest.restoreAllMocks();

    const store = buildStore();
    const company = store.searchResults.find(el => el.status === "COMPLETED");
    const component = mount(
      <Result store={store} key={company.company} company={company} />
    );
    expect(store.activeJobs.length).toBe(2);
    expect(store.ignoredJobs.length).toBe(2);

    jest.spyOn(global, "fetch").mockImplementation(() => {
      return new Promise((resolve, reject) => {
        resolve(
          new Response(
            JSON.stringify([
              {
                _id: "GHI123",
                title: "Job Title GHI123",
                location: "San Francisco",
                url: "https://www.example.com/job/GHI123"
              }
            ])
          )
        );
      });
    });

    component
      .find("button.watchButton")
      .first()
      .simulate("click");

    setTimeout(() => {
      component.update();
      expect(store.activeJobs.length).toBe(3);
      expect(store.ignoredJobs.length).toBe(2);

      expect(component.find("button.archiveButton").length).toBe(2);
      expect(component.find("button.watchButton").length).toBe(1);

      expect(component.find("div.activeJobs a").length).toBe(3); //3 'Active' jobs
      expect(component.find("div.ignoredJobs a").length).toBe(1); //1 'Ignored' job
      expect(component.find("div.otherJobs a").length).toBe(1); //1 'Other' job

      expect(component).toMatchSnapshot();

      expect(global.fetch).toHaveBeenCalledTimes(1);
      expect(global.fetch).toHaveBeenCalledWith(
        "/job/add-job",
        expect.anything()
      );
      global.fetch.mockClear();

      done();
    });
  });

  test("Render result for good results, ignore one", done => {
    jest.restoreAllMocks();

    const store = buildStore();
    const company = store.searchResults.find(el => el.status === "COMPLETED");
    const component = mount(
      <Result store={store} key={company.company} company={company} />
    );
    expect(store.activeJobs.length).toBe(2);
    expect(store.ignoredJobs.length).toBe(2);

    jest.spyOn(global, "fetch").mockImplementation(() => {
      return new Promise((resolve, reject) => {
        resolve(
          new Response(
            JSON.stringify([
              {
                _id: "JKL123",
                title: "Job Title JKL123",
                location: "San Francisco",
                url: "https://www.example.com/job/JKL123"
              }
            ])
          )
        );
      });
    });

    component
      .find("button.ignoreButton")
      .last()
      .simulate("click");

    setTimeout(() => {
      component.update();

      expect(store.activeJobs.length).toBe(2);
      expect(store.ignoredJobs.length).toBe(3);
      expect(component.find("button.archiveButton").length).toBe(1);
      expect(component.find("button.watchButton").length).toBe(1);

      expect(component.find("div.activeJobs a").length).toBe(2); //2 'Active' jobs, we ignored one
      expect(component.find("div.ignoredJobs a").length).toBe(2); //2 'Ignored' jobs, we ignored one
      expect(component.find("div.otherJobs a").length).toBe(1); //1 'Other' job

      expect(component).toMatchSnapshot();

      expect(global.fetch).toHaveBeenCalledTimes(1);
      expect(global.fetch).toHaveBeenCalledWith(
        "/job/ignore-job",
        expect.anything()
      );
      global.fetch.mockClear();

      done();
    });
  });
});
