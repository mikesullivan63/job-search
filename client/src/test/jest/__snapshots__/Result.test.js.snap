// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Display tests Render result for completed results 1`] = `ReactWrapper {}`;

exports[`Result Display tests Render result for error result 1`] = `
<div
  className="ui red inverted secondary segment"
>
  <a
    href="https://www.error.com/job/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <h4>
      Error.com
    </h4>
  </a>
  <span>
    Problem accessing page
  </span>
</div>
`;

exports[`Result Display tests Render result for good results, archive one of them 1`] = `ReactWrapper {}`;

exports[`Result Display tests Render result for good results, ignore one 1`] = `ReactWrapper {}`;

exports[`Result Display tests Render result for good results, watch one 1`] = `ReactWrapper {}`;

exports[`Result Display tests Render result for pending result 1`] = `
<div
  className="ui tertiary segment"
>
  <a
    href="https://www.example.io/job/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <h4>
      Example.io
    </h4>
  </a>
  <span>
    Pending Lookup
  </span>
</div>
`;
